### ISSUE TITLE
Should make_tempfile use realpath?

### ISSUE THREAD

**Issue by kyleam** [datalad/datalad#2640]
`datalad.utils.make_tempfile` calls `os.path.realpath` on the generated path:

https://github.com/datalad/datalad/blob/973f9f539510314570a2ab5dd17d04cef7945dc4/datalad/utils.py#L1404-L1406

This means that `datalad.tests.utils.with_tempfile` resolves any symlinks in its returned path.  In contrast, `datalad.tests.utils.with_tree` does **not** resolve symlinks.  This leads to confusing (at least to me) situations like the one in b4b310458 (more [here]), where a test with TMPDIR="/var/tmp/sym link" will pass when `with_tempfile` is used and fail when `with_tree` is used.

So I wonder if we should drop the `realpath` in `make_tempfile`

  * so that `with_tempfile` behaves the same as `with_tree`

  * to see if using `with_tempfile` is masking any failures in the TMPDIR="/var/tmp/sym link" case.

But I don't have a good understanding of this issue because a86f5756f specifically added the `realpath` call to deal with test issues.

[here]: https://github.com/datalad/datalad/pull/2634#issuecomment-395847882

**Comment by yarikoptic** [comment]
hm, good find.  I thought that I've raided all realpaths and added comments where it was "ok".  I do not see why here we would want to use realpath indeed, and I didn't know that we did.  I bet if it gets abolished we might unravel a few "symlink in the path" related issues.

The https://github.com/datalad/datalad/commit/a86f5756fdbd2e9549b990bbea0c345634e6f939 you refer to where it was added in 2015 when the world was still flat ;-)
